tests/advectionBackwardEuler.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Jun  4 13:31:10 2021
+
+@author: nikol
+"""
+
+from fenics import * 
+
+from irksome import GaussLegendre, RadauIIA, Dt, TimeStepper
+
+
+  
+from ufl.algorithms.ad import expand_derivatives
+
+import numpy as np
+
+
+msh = IntervalMesh(10, 0, 1)
+
+V = FunctionSpace(msh, "CG", 1)
+
+
+  
+T = 2.0            # final time
+num_steps = 20     # number of time steps
+dt = T / num_steps # time step size
+
+ 
+c= Constant(2.0)
+u_D = Expression('x[0]-c*t', degree=1, c=c,t=0)
+        
+def boundary(x, on_boundary):
+            return on_boundary
+        
+bc = DirichletBC(V, u_D, boundary)
+
+
+u_n = interpolate(u_D, V)
+u = TrialFunction(V)
+k0= Function(V)
+v0=TestFunction(V)
+u0= u_n + dt*Constant(1.0)*k0     
+F = u*v0*dx+u0*v0*dx- dt*c*Dx(u0,0)*v0*k0*dx
+a, L = lhs(F), rhs(F)
+
+
+
+u = Function(V)
+t = 0
+arrayY=[]
+arrayX=[]
```

Comment:
You can also use the output vtk functionality from FEniCS and visualize with paraview:

```suggestion
out = File("out.pvd")
out << u_n
```

For me this looks the following way:

![Screenshot from 2021-06-10 19-21-24](https://user-images.githubusercontent.com/5740604/121569437-13cf0d00-ca21-11eb-8266-78e94dfdcc2b.png)